File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    const clothObj = clothes.reduce((obj, [,category]) => {
        obj[category] ??= 0
        obj[category]++
        return obj
    }, {})
    
    return Object.entries(clothObj).reduce((res, [,count]) => res * (count + 1), 1) - 1
    
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
import static java.util.stream.Collectors.*;

class Solution {
    public int solution(String[][] clothes) {
        return Arrays.stream(clothes)
            .collect(
                groupingBy(
                    p -> p[1],
                    mapping(p -> p[0], counting())
                )
            )
            .values()
            .stream()
            .reduce(1L, (acc, cur) -> acc * (cur + 1))
            .intValue() - 1;
    }
}